gateway/src/main/java/org/georchestra/gateway/filter/headers/RemoveHeadersGatewayFilterFactory.java

```diff
@@ -107,17 +107,21 @@ private Pattern pattern() {
             return compiled;
         }
 
-        boolean matches(@NonNull String headerName) {
-            return pattern().matcher(headerName).matches();
+        boolean anyMatches(@NonNull HttpHeaders httpHeaders) {
+            return httpHeaders.keySet().stream().anyMatch(h -> this.matches(h, httpHeaders.get(h)));
+        }
+
```

Comment:
I added this method because it was used in the u-test, but I am wondering if is actually necessary.